packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,494 @@
+
+As someone who is making an FCL compatible wallet, you will need to make and expose a webpage hosted at an authentication endpoint that will be rendered in an iframe.
+
+```javascript
+// IN APPLICATION
+// configuring fcl to point at a wallet looks like this
+import {config} from "@onflow/fcl"
+
+config({
+  "discovery.wallet": "your-url-that-fcl-will-render-in-iframe"
+})
+```
+
+Once the page is rendered, the wallet then needs to tell FCL that it is ready. You will do this by sending a post message to FCL, and FCL will send back a post message with some additional information that you can use about the application requesting authentication on behalf of the user.
+
+```javascript
+// IN WALLET AUTHENTICATION FRAME
+
+function callback({ data }) {
+  // early exit if it isnt the FCL:AUTHN:CONFIG message
+  if (typeof data != "object") return
+  if (typeof data.type !== "FCL:AUTHN:CONFIG") return
+
+  // this might have a bunch of usefull information  
+  doSomethingWithConfigData(e.data)
+
+  // always clean up your eventlisteners
+  window.removeEventListener(callback)
+}
+// add event listener first
+window.addEventListener("message", callback)
+
+// tell fcl the wallet is ready
+window.parent.postMessage({type: "FCL:FRAME:READY"}, "*")
+```
+
+You will learn fairly fast that almost everything in FCL is optional. The config that FCL sends here is the applications chance to suggest to you, the wallet, what they would like you to send back to them.
+
+An example of that is the OpenID service. The application can request for you, the wallet, to send them the email address of the current user. The application does this through the config highlighted in the example above. The application requesting this information does not mean you need to send it. It's entirely optional for you, the wallet, to do so.
+
+In the config they can also tell you, the wallet, a variety of things about them, the application, such as the name of their application or a url for an icon of their application. You can use these pieces of information to customize your wallets user experience should you desire to do so.
+
+You, the wallet, having a visual distinction from the application, but still a seemless and connected experience is our goal here.
+
+As always, you must never trust anything you recieve from an application. Always do your due-dilligence and be onguard as you, the wallet, are the users first line of defense against potentially malicious applications.
+
+### Authenticate your User 
+
+It's important that you, the wallet, are confident that the user is the user.
+
+Have them provide enough proof to you that you are okay with passing their details back to FCL.
+If we were to use Blocto as an example, this is when the user is logging into Blocto, putting in their email and getting sent an authentication code to their email. Everything Blocto needs to do to be confident in the users identity.
```

Comment:
Thinking this needs an edit, second part is not a sentence...

_Using Blocto as an example, an authentication code is sent to the email a user enters at login. 
This code can be used as validation and is everything Blocto needs to be confident in the users identity._